### PR TITLE
GAT-5830 :: create and update secrets

### PR DESCRIPTION
**Pull request by dnhdruk**
## Screenshots (if relevant)
Screenshot 2024-12-11 at 14 21 41
Screenshot 2024-12-11 at 14 21 51

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added appropriate unit tests
- [ ] I have created mocks for unit tests (where appropriate)
- [ ] I have added appropriate Behat tests to confirm AC (if applicable)
- [ ] I have added Swagger annotations for new endpoints (if applicable)
- [ ] I have added audit logs for new operation logic (if applicable)
- [ ] I have added new environment variables to the .env.example file (if applicable)
- [ ] I have added new environment variables to terraform repository (if applicable)

### CODE REVIEW

File: app/Models/Federation.php [hunk 1]
```diff
@@ -17,16 +16,6 @@ class Federation extends Model
...
 
-    protected static function boot()
-    {
-        parent::boot();
-
-        //create a pid for this team
-        static::creating(function ($model) {
-            $model->pid = (string) Str::uuid();
-        });
-    }
```
Comment (loki-sinclair-hdruk) [inline app/Models/Federation.php:28]: This removal of this, isn't required. In fact, the only part missing was the addition to $fillable on the model. Without that being set, Laravel doesn't know what to do with the extra param. With it set, `boot` will be called first, so this is perfectly valid and then doesn't require updating the controllers above.

Comment (dnhdruk) [reply app/Models/Federation.php:28]: I think `pid` is not necessary because `id` is enough for uniqueness.

Comment (dnhdruk) [reply app/Models/Federation.php:28]: updated based on `id`

Comment (loki-sinclair-hdruk) [reply app/Models/Federation.php:28]: Fair enough, your call. :)

File: rest.http [hunk 2]
```diff
@@ -21,6 +21,7 @@
 # 'developers@hdruk.ac.uk', 'Watch26Task?'
```
Comment (loki-sinclair-hdruk) [inline rest.http]: Sorry, missed this. We _really_ need to get rid of these plain text credentials.

Comment (dnhdruk) [reply rest.http]: updated

**Review by loki-sinclair-hdruk — APPROVED** [review]

**Review by BSnelling — APPROVED** [review]

**Pull request merged**